### DISCUSSION TITLE
prog_models: Module not found

### DISCUSSION THREAD

**Discussion by bkramesh64** [nasa/prog_models#582] [General]
Running future_loading.py example that has
"""
Example demonstrating ways to use future loading. 
"""

import matplotlib.pyplot as plt
from numpy.random import normal
**from prog_models.loading import Piecewise, GaussianNoiseLoadWrapper**
from prog_models.models import BatteryCircuit
from statistics import mean
getting " ModuleNotFoundError: No module named 'prog_models.loading'

**Comment by teubert** [comment]
Hello @bkramesh64,

I'm sorry to hear about your trouble. I'll do what I can to help you resolve it. 

It seems that python either cannot find your installation of prog_models or cannot find the loading sub package. Just to confirm- have you installed the prog_models package (pip install prog_models)?

**Comment by bkramesh64** [comment]
Hello Christopher

Yes, I have Pip installed prog_models.  I use macos High Sierra 10.13.6.
Python 3.10, Spyder IDE to run the python code.

Regards
Ramesh

**Reply by teubert** [reply]
Ok, can you tell me what version of prog_models you installed?

You can see this by running 'pip list'

**Comment by bkramesh64** [comment]
I have installed prog_models 1.3.1

**Reply by teubert** [reply]
I think that's your issue. The loading sub package was introduced in v1.5. Could you try updating your installation of prog_models and let me know if that fixes it?

This command should update it:
'pip install prog_models --upgrade'

**Comment by bkramesh64** [comment]
Thank you Christopher.  I tried upgrading it but somehow it didn't update
to 1.5. I need to try with Pip3 or update pip.
I will try and I guess it should help.

**Comment by bkramesh64** [comment]
Hi Chris

For me pip install prog_models --upgrade is only installing 1.3.1  and not
able to install 1.5.  Pl find the details below:

pip install prog_models --upgrade

Collecting prog_models

  Obtaining dependency information for prog_models from
https://files.pythonhosted.org/packages/fa/af/ba671886582d3f3b3c08ad95246a901c47684d6bb6a77ba4740fda86e905/prog_models-1.5.2-py3-none-any.whl.metadata

  Using cached prog_models-1.5.2-py3-none-any.whl.metadata (7.3 kB)

Requirement already satisfied: scipy in
/anaconda3/envs/bkenv/lib/python3.10/site-packages (from prog_models)
(1.9.3)

Requirement already satisfied: pandas in
/anaconda3/envs/bkenv/lib/python3.10/site-packages (from prog_models)
(1.5.2)

Requirement already satisfied: matplotlib in
/anaconda3/envs/bkenv/lib/python3.10/site-packages (from prog_models)
(3.3.1)

Requirement already satisfied: requests in
/anaconda3/envs/bkenv/lib/python3.10/site-packages (from prog_models)
(2.28.1)

Collecting chaospy (from prog_models)

  Obtaining dependency information for chaospy from
https://files.pythonhosted.org/packages/a1/6f/d4e7914cdb03712bb8a5a0d06be929047ea8c2a7a44f3f607d1e8f27a4bc/chaospy-4.3.13-py3-none-any.whl.metadata

  Using cached chaospy-4.3.13-py3-none-any.whl.metadata (5.6 kB)

Collecting fastdtw (from prog_models)

  Using cached fastdtw-0.3.4.tar.gz (133 kB)

  Preparing metadata (setup.py) ... done

Requirement already satisfied: tensorflow in
/anaconda3/envs/bkenv/lib/python3.10/site-packages (from prog_models)
(2.10.0)

Requirement already satisfied: numpy>=1.20 in
/anaconda3/envs/bkenv/lib/python3.10/site-packages (from
chaospy->prog_models) (1.23.4)

Collecting numpoly>=1.2.5 (from chaospy->prog_models)

  Obtaining dependency information for numpoly>=1.2.5 from
https://files.pythonhosted.org/packages/63/cb/58b864a0cd9fc67d8114e922ce734821da31bd27f01f076b22df47c07324/numpoly-1.2.11-py3-none-any.whl.metadata

  Using cached numpoly-1.2.11-py3-none-any.whl.metadata (6.0 kB)

Requirement already satisfied: setuptools>=40.9.0 in
/anaconda3/envs/bkenv/lib/python3.10/site-packages (from
chaospy->prog_models) (65.5.0)

Requirement already satisfied: importlib-metadata in
/anaconda3/envs/bkenv/lib/python3.10/site-packages (from
chaospy->prog_models) (4.13.0)

Requirement already satisfied: certifi>=2020.06.20 in
/anaconda3/envs/bkenv/lib/python3.10/site-packages (from
matplotlib->prog_models) (2022.9.24)

Requirement already satisfied: cycler>=0.10 in
/anaconda3/envs/bkenv/lib/python3.10/site-packages (from
matplotlib->prog_models) (0.11.0)

Requirement already satisfied: kiwisolver>=1.0.1 in
/anaconda3/envs/bkenv/lib/python3.10/site-packages (from
matplotlib->prog_models) (1.4.4)

Requirement already satisfied: pillow>=6.2.0 in
/anaconda3/envs/bkenv/lib/python3.10/site-packages (from
matplotlib->prog_models) (9.3.0)

Requirement already satisfied: pyparsing!=2.0.4,!=2.1.2,!=2.1.6,>=2.0.3 in
/anaconda3/envs/bkenv/lib/python3.10/site-packages (from
matplotlib->prog_models) (3.0.9)

Requirement already satisfied: python-dateutil>=2.1 in
/anaconda3/envs/bkenv/lib/python3.10/site-packages (from
matplotlib->prog_models) (2.8.2)

Requirement already satisfied: pytz>=2020.1 in
/anaconda3/envs/bkenv/lib/python3.10/site-packages (from
pandas->prog_models) (2022.7)

Requirement already satisfied: charset-normalizer<3,>=2 in
/anaconda3/envs/bkenv/lib/python3.10/site-packages (from
requests->prog_models) (2.0.4)

Requirement already satisfied: idna<4,>=2.5 in
/anaconda3/envs/bkenv/lib/python3.10/site-packages (from
requests->prog_models) (3.4)

Requirement already satisfied: urllib3<1.27,>=1.21.1 in
/anaconda3/envs/bkenv/lib/python3.10/site-packages (from
requests->prog_models) (1.26.13)

Requirement already satisfied: absl-py>=1.0.0 in
/anaconda3/envs/bkenv/lib/python3.10/site-packages (from
tensorflow->prog_models) (1.3.0)

Requirement already satisfied: astunparse>=1.6.0 in
/anaconda3/envs/bkenv/lib/python3.10/site-packages (from
tensorflow->prog_models) (1.6.3)

Requirement already satisfied: flatbuffers>=2.0 in
/anaconda3/envs/bkenv/lib/python3.10/site-packages (from
tensorflow->prog_models) (2.0)

Requirement already satisfied: gast<=0.4.0,>=0.2.1 in
/anaconda3/envs/bkenv/lib/python3.10/site-packages (from
tensorflow->prog_models) (0.4.0)

Requirement already satisfied: google-pasta>=0.1.1 in
/anaconda3/envs/bkenv/lib/python3.10/site-packages (from
tensorflow->prog_models) (0.2.0)

Requirement already satisfied: h5py>=2.9.0 in
/anaconda3/envs/bkenv/lib/python3.10/site-packages (from
tensorflow->prog_models) (3.7.0)

Collecting keras-preprocessing>=1.1.1 (from tensorflow->prog_models)

  Using cached Keras_Preprocessing-1.1.2-py2.py3-none-any.whl (42 kB)

Collecting libclang>=13.0.0 (from tensorflow->prog_models)

  Obtaining dependency information for libclang>=13.0.0 from
https://files.pythonhosted.org/packages/c9/ea/fe2a69cc6cfebf7c7ee8a6357566fc1cbb91632bde5869b669a396accb5f/libclang-16.0.6-py2.py3-none-macosx_10_9_x86_64.whl.metadata

  Using cached libclang-16.0.6-py2.py3-none-macosx_10_9_x86_64.whl.metadata
(5.2 kB)

Requirement already satisfied: opt-einsum>=2.3.2 in
/anaconda3/envs/bkenv/lib/python3.10/site-packages (from
tensorflow->prog_models) (3.3.0)

Requirement already satisfied: packaging in
/anaconda3/envs/bkenv/lib/python3.10/site-packages (from
tensorflow->prog_models) (22.0)

Collecting protobuf<3.20,>=3.9.2 (from tensorflow->prog_models)

  Using cached protobuf-3.19.6-py2.py3-none-any.whl (162 kB)

Requirement already satisfied: six>=1.12.0 in
/anaconda3/envs/bkenv/lib/python3.10/site-packages (from
tensorflow->prog_models) (1.16.0)

Requirement already satisfied: termcolor>=1.1.0 in
/anaconda3/envs/bkenv/lib/python3.10/site-packages (from
tensorflow->prog_models) (2.1.0)

Requirement already satisfied: typing-extensions>=3.6.6 in
/anaconda3/envs/bkenv/lib/python3.10/site-packages (from
tensorflow->prog_models) (4.4.0)

Requirement already satisfied: wrapt>=1.11.0 in
/anaconda3/envs/bkenv/lib/python3.10/site-packages (from
tensorflow->prog_models) (1.14.1)

INFO: pip is looking at multiple versions of tensorflow to determine which
version is compatible with other requirements. This could take a while.

Collecting prog_models

  Obtaining dependency information for prog_models from
https://files.pythonhosted.org/packages/38/4b/39698817375d9592118091040bc32c874152f04affc7a6c042e09f178c11/prog_models-1.5.1-py3-none-any.whl.metadata

  Using cached prog_models-1.5.1-py3-none-any.whl.metadata (7.3 kB)

  Obtaining dependency information for prog_models from
https://files.pythonhosted.org/packages/36/5f/7e21d4818e5c1eeb3d3596f8fd056b888449e993314a0ca7d4634ebc65bf/prog_models-1.5.0-py3-none-any.whl.metadata

  Using cached prog_models-1.5.0-py3-none-any.whl.metadata (7.3 kB)

  Using cached prog_models-1.4.6-py3-none-any.whl (179 kB)

  Using cached prog_models-1.4.5-py3-none-any.whl (179 kB)

  Using cached prog_models-1.4.4-py3-none-any.whl (179 kB)

  Using cached prog_models-1.4.3-py3-none-any.whl (179 kB)

  Using cached prog_models-1.4.2-py3-none-any.whl (179 kB)

INFO: pip is looking at multiple versions of prog-models to determine which
version is compatible with other requirements. This could take a while.

  Downloading prog_models-1.3.1-py3-none-any.whl (156 kB)

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 156.2/156.2 kB 1.1 MB/s eta
0:00:00

Installing collected packages: prog_models

Successfully installed prog_models-1.3.1
